lib/Sema/CodeSynthesis.cpp

```diff
@@ -2115,14 +2115,16 @@ static void maybeAddMemberwiseDefaultArg(ParamDecl *arg, VarDecl *var,
   // default arg. All lazy variables return a nil literal as well. *Note* that
   // the type will always be a sugared T? because we don't default init an
   // explicit Optional<T>.
+  bool isNilInitialized =
+    var->getAttrs().hasAttribute<LazyAttr>() ||
+    (!isExplicitlyInitialized && isDefaultInitializable &&
+     var->getValueInterfaceType()->getAnyNominal() == ctx.getOptionalDecl() &&
```

Comment:
I would write this as var->getValueInterfaceType()->getOptionalObjectType()

Reply:
Ah, good call. I'll do that on master.